storage/ndb/rest-server2/server/src/metadata.cpp

```diff
@@ -59,56 +59,53 @@ bool FeatureMetadata::isComplex() const {
 
 AvroDecoder::AvroDecoder() = default;
 
-AvroDecoder::AvroDecoder(const std::string &schemaJson) {
-  schema = avro::compileJsonSchemaFromString(schemaJson);
+AvroDecoder::~AvroDecoder() {
 }
 
-std::pair<RS_Status, std::optional<avro::GenericDatum>>
-  AvroDecoder::decode(const std::vector<Uint8> &inData) const {
-
-  if (!schema.root()) {
-    return {CRS_Status(HTTP_CODE::SERVER_ERROR, "Invalid avro schema").status, std::nullopt} ;
+void AvroDecoder::unregister_with_go_layer() {
+  // unregister_schema
+  if (schemaID != -1) {
+    DEB_MD_CACHE("Removing Schemd %s. ID: %d from the golang layer",
+      schemaStr, schemaID);
+    unregister_schema(schemaID);
+    schemaID = -1;
   }
+}
 
-  auto inStream = avro::memoryInputStream(inData.data(), inData.size());
-  avro::DecoderPtr decoder = avro::binaryDecoder();
-  if(!decoder) {
-    return {CRS_Status(HTTP_CODE::SERVER_ERROR, "Avro failed to decode data").status, std::nullopt};
+RS_Status AvroDecoder::register_with_go_layer() {
+  GoString goString;
+  goString.p = schemaJson.c_str();
+  goString.n = schemaJson.length();
+  int64_t id = register_schema(goString);
+  if (id == -1) {
+   return CRS_Status(HTTP_CODE::SERVER_ERROR, "Failed to register avro schema").status;
```

Comment:
Returning a simple error code and creating more elaborate error codes in HTTP response seems to be the way to go, rather than clogging the log